sharepoint/3.2/modules/ROOT/pages/sharepoint-connector-reference.adoc

```diff
@@ -0,0 +1,1775 @@
+| Configuration | String | Name of the configuration to use | | x
+| File Server Relative Url a| String | The path of the file to be deleted |  | x
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
+|===
+
+
+=== Associated Configurations
+* <<sharepoint-online>>
+* <<sharepoint-onpremise>>
+
+==== Throws
+* SHAREPOINT:RETRY_EXHAUSTED
+* SHAREPOINT:UNKNOWN
+* SHAREPOINT:CONNECTIVITY
+
+
+[[fileDeny]]
+=== File Deny
+`<sharepoint:file-deny>`
+
+
+Denies approval for a file that was submitted for content approval. Only documents in lists that are enabled for content approval can be denied.
+
+
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use | | x
+| File Server Relative Url a| String | The path of the file to be denied |  | x
+| Comment a| String | Additional comment for this deny operation |  |
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
+|===
+
+
+=== Associated Configurations
+* <<sharepoint-online>>
+* <<sharepoint-onpremise>>
+
+==== Throws
+* SHAREPOINT:RETRY_EXHAUSTED
+* SHAREPOINT:UNKNOWN
+* SHAREPOINT:CONNECTIVITY
+
+
+[[fileGetContent]]
+=== File Get Content
+`<sharepoint:file-get-content>`
+
+
+Retrieves content of a file in SharePoint.
+
+
+==== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| File Server Relative Url a| String | Target file to retrieve its content from SharePoint. |  | x
+| Target Variable a| String |  The name of a variable that stores the operation's output. |  |
+| Target Value a| String |  An expression that evaluates against the operation's output. The outcome of this expression is stored in the target variable. |  #[payload] |
+| Reconnection Strategy a| * <<reconnect>>
+* <<reconnect-forever>> |  A retry strategy in case of connectivity errors |  |
+|===
+
+==== Output
+[%autowidth.spread]
+|===
+|Type |Binary
+|===
+
+=== Associated Configurations
+* <<sharepoint-online>>
+* <<sharepoint-onpremise>>
+
+==== Throws
+* SHAREPOINT:RETRY_EXHAUSTED
+* SHAREPOINT:UNKNOWN
+* SHAREPOINT:CONNECTIVITY
+
+
+[[fileMove]]
+=== File Move
+`<sharepoint:file-move>`
+
+Executes an HTTP POST against the Sharepoint API to move a file. The parameter fileServerRelativeUrl needs to point to an existing file in SharePoint. The parameter newFileServerRelativeUrl should be a path to where the file should be moved. If this is a directory, an error will be thrown since it should be a new file path. It can have the same name as the old file or it can have a new name. In order to use this operation as just a rename, specify the same path in both parameters and use a different file name in newFileServerRelativeUrl.
```

Comment:
```suggestion
Executes an HTTP POST against the Sharepoint API to move a file. The `fileServerRelativeUrl` parameter must point to an existing file in SharePoint. The `newFileServerRelativeUrl` parameter must be the path for where to move the file. If the provided path is to a directory, an error will be thrown since it should be a new file path. The new file can have the same name as the old file or it can have a new name. 

To use this operation as a rename operation, specify the same path in both parameters and use a different file name in the `newFileServerRelativeUrl` parameter.
```